### PR TITLE
Use JSON as the wire format in Gearman jobs

### LINKED ISSUE #1673: Problem: use of pickle for data serialization is disproportionate
**Expected behaviour**
Archivematica should use a serialization format that is both safe and interoperable. Doing so would enable the development of safer and more extensible systems using tools outside the Python ecosystem, given that pickle is Python-specific.

**Current behaviour**
Archivematica uses the [pickle](https://docs.python.org/3/library/pickle.html) module for serializing and de-serializing Python objects in the following scenarios:

<details>
<summary>RPC server (MCPServer, Dashboard)</summary>
<pre>
- src/MCPServer/lib/server/rpc_server.py
- src/dashboard/src/contrib/mcp/client.py
- src/dashboard/src/components/mcp/views.py
- tests/dashboard/test_api.py
- tests/dashboard/components/mcp/test_views.py
</pre>
</details>

<details>
<summary>Task backend (MCPServer, MCPClient)</summary>
<pre>
- src/MCPServer/lib/server/tasks/backends/gearman_backend.py
- src/MCPClient/lib/client/gearman.py
- tests/MCPServer/test_gearman.py
</pre>
</details>

<details>
<summary>Access details (MCPClient, Dashboard)</summary>
<pre>
- src/MCPClient/lib/clientScripts/upload_qubit.py
- tests/MCPClient/test_upload_qubit.py
- src/dashboard/src/components/ingest/views.py
- tests/dashboard/components/mcp/test_views.py
- tests/dashboard/test_ingest.py
</pre>
</details>

In all the cases listed above, the JSON serialization format would suffice given that we don't make use of the more advance capabilities supported by the pickle module. See the [comparison with JSON](https://docs.python.org/3/library/pickle.html#comparison-with-json) for more details.

Regarding performance, we are currently using pickle's protocol version 0 which is slower compared to the JSON encoder included in the standard library. If we choose to use JSON and run into performance issues down the road, [orjson](https://github.com/ijl/orjson) is a faster alternative.

**Steps to reproduce**
N/A

**Your environment (version of Archivematica, operating system, other relevant details)**
All known versions of Archivematica.

---

**For Artefactual use:**

Before you close this issue, you must check off the following:

- [ ] All pull requests related to this issue are properly linked
- [ ] All pull requests related to this issue have been merged
- [ ] A testing plan for this issue has been implemented and passed (testing plan information should be included in the issue body or comments)
- [ ] Documentation regarding this issue has been written and merged (if applicable)
- [ ] Details about this issue have been added to the release notes (if applicable)

### PR DESCRIPTION
**Pull request by sevein**
This pull request replaces all uses of `pickle` with `json`. The use of `pickle` was [introduced back in 2011](https://github.com/artefactual/archivematica-history/commit/79a5ff1ada48d579f7b295ab0dad212bfad74250) when MCP was integrated with Gearman due to its ability to round-trip Python objects, at the cost of lower security. However, apart from `datetime.datetime` and `uuid.UUID`, we are not encoding any complex structures that cannot be easily represented in JSON.

Please notice that the use of `orjson` was an afterthought and that's why it was submitted as an extra commit. `orjson` is significantly faster compared to the standard library `json` module, being faster than pickle protocol=0 that we were using previously. It also supports native serialization of datetime and UUID objects.

Fixes https://github.com/archivematica/Issues/issues/1673.

### CODE REVIEW

File: hack/docker-compose.yml [hunk 1]
```diff
@@ -39,6 +39,8 @@ services:
       - "mysql_data:/var/lib/mysql"
     ports:
       - "127.0.0.1:62001:3306"
+    cap_add:
+      - "SYS_NICE"
```
Comment (replaceafill) [inline hack/docker-compose.yml]: Why is this necessary in the `mysql` service?

**Review by replaceafill — APPROVED** [review]
This looks great @sevein, thank you!

I think this makes the interaction with Gearman clearer and easier to debug :bow: and I like that the use of `orjson` is nicely abstracted in a single module.

File: hack/docker-compose.yml (hunk 1 above)
Comment (sevein) [reply hack/docker-compose.yml]: I was seeing this line going forever in my logs:

    mbind: Operation not permitted

I've removed the commit, will look at it again if it comes back.

Comment (replaceafill) [reply hack/docker-compose.yml]: I just set up the latest `qa/*` branches and fresh containers and I see the message in the logs now:

```
mysql-1  | 2024-05-24T21:37:45.709752Z 0 [System] [MY-011323] [Server] X Plugin ready for connections. Bind-address: '::' port: 33060, socket: /var/lib/mysql/mysqlx.sock
mysql-1  | 2024-05-24T21:37:45.709838Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.36-28'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  Percona Server (GPL), Release 28, Revision 47601f19.
mysql-1  | mbind: Operation not permitted
mysql-1  | mbind: Operation not permitted
mysql-1  | mbind: Operation not permitted
mysql-1  | mbind: Operation not permitted
mysql-1  | mbind: Operation not permitted
mysql-1  | mbind: Operation not permitted
mysql-1  | mbind: Operation not permitted
mysql-1  | mbind: Operation not permitted
mysql-1  | mbind: Operation not permitted
mysql-1  | mbind: Operation not permitted
mysql-1  | mbind: Operation not permitted
```
